### DISCUSSION TITLE
How do I rollback a migration?

### DISCUSSION THREAD

**Discussion by fmh-finanzberatung** [keystonejs/keystone#8659] [Questions] [answered]
I need to be able to rollback migrations. How do I do that?

Otherwise  I end up with tons of migrations during development.

**Comment by marekryb** [comment, marked as answer]
With anything related to migrations, you are dealing directly with [Prisma migrate](https://www.prisma.io/docs/guides/migrate).
See related docs (not sure you actually mean rollback - seems you need squash):
[Squashing migrations](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/squashing-migrations)
[Generating down migrations](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/generating-down-migrations)

**Reply by fmh-finanzberatung** [reply]
Awesome, exactly what I was looking for. I've searched up and down the web for that "Generating down" and "Squashing" migrations functionality. Thanks!
